src/course/infrastructure/controller/express-course-controller.ts

```diff
@@ -0,0 +1,139 @@
+import { Request, Response } from 'express';
+import { services } from '../../../share/infrastructure/services';
```

Comment:
Controller imports a global 'services' object (../../../share/infrastructure/services) and uses it directly. This couples the controller to a global service locator and violates Hexagonal principles (controller should depend on ports/interfaces injected into it).

**Sugerencia:**
```typescript
Inject needed application services (use-cases) through the controller's constructor so tests and other adapters can provide mocks. Example:
export class ExpressCourseController {
  constructor(private services: { courseGetAll: CourseGetAll; courseCreate: CourseCreate; ... }) {}
  // use this.services.courseGetAll.execute(...) instead
}
Or use a DI container at app bootstrap to wire dependencies, but avoid importing a global services object inside the controller.
```